app/org/sagebionetworks/bridge/services/AuthenticationServiceImpl.java

```diff
@@ -246,14 +257,23 @@ private UserSession getSessionFromAccount(final Study study, final Account accou
 
         final String healthCode = getHealthCode(study, account);
         user.setHealthCode(healthCode);
-
+        
         user.setSharingScope(optionsService.getEnum(healthCode, SHARING_SCOPE, SharingScope.class));
         user.setDataGroups(optionsService.getStringSet(healthCode, DATA_GROUPS));
-        
-        user.setSignedMostRecentConsent(consentService.hasUserSignedActiveConsent(study, user));
-        user.setConsent(consentService.hasUserConsentedToResearch(study, user));
 
+        // now that we know more about this user, we can expand on the request context.
+        context = new ScheduleContext.Builder()
+                .withContext(context)
+                .withHealthCode(healthCode)
+                .withUserDataGroups(user.getDataGroups())
+                .withStudyIdentifier(study.getIdentifier()) // probably already set
+                .build();
```

Comment:
Similarly, there are a lot of fields in ScheduleContext that aren't needed to get consent. This suggests to me that we should have a separate "ConsentContext" class that contains only the fields needed for consents.